### PR TITLE
Add new entries to clarify multicluster terminology.

### PR DESCRIPTION
**Pull request by rcaballeromx**
These entries are part of a wider effort to improve
multicluster content.  Implementing the terms
across the current content will happen in a separate PR.

Signed-off-by: Rigs Caballero <grca@google.com>

[ ] Configuration Infrastructure
[ X ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/reference/glossary/primary-cluster.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: Primary Cluster
+test: n/a
+---
+
+A primary cluster is a cluster with a control plane, which observes all
+clusters in the mesh for endpoints and facilitates cross-cluster load balancing
```
Comment (frankbu) [inline content/en/docs/reference/glossary/primary-cluster.md:7]: Is this correct?
```suggestion
A primary cluster is a cluster with a control plane, which observes more than one
cluster in a multicluter mesh for endpoints and facilitates cross-cluster load balancing
```

Comment (rcaballeromx) [reply content/en/docs/reference/glossary/primary-cluster.md:7]: @nmittler @smawson WDYT?

Comment (nmittler) [reply content/en/docs/reference/glossary/primary-cluster.md:7]: I think the original text was correct. A single-cluster mesh consists of a single primary cluster.

Comment (frankbu) [reply content/en/docs/reference/glossary/primary-cluster.md:7]: I was worried about the other end, "observes all clusters". Is that always true? Isn't it possible to have a primary cluster that only observes a subset of the clusters in the mesh, while another primary cluster handles others?

Comment (nmittler) [reply content/en/docs/reference/glossary/primary-cluster.md:7]: Not in any documented configuration that I'm aware of. Unless you're talking about versioned control planes, which is slightly orthogonal.

File: content/en/docs/reference/glossary/primary-cluster.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+title: Primary Cluster
+test: n/a
+---
+
+A primary cluster is a cluster with a control plane, which observes all
```
Comment (smawson) [inline content/en/docs/reference/glossary/primary-cluster.md:6]: Can/should we link to the glossary entry for control plane here?

File: content/en/docs/reference/glossary/primary-cluster.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+clusters in the mesh for endpoints and facilitates cross-cluster load balancing
+between clusters.
```
Comment (smawson) [inline content/en/docs/reference/glossary/primary-cluster.md:8]: Things we should mention:
- Multiple primary clusters can be connected together in a single mesh (point to glossary entry for mesh)
- Primary clusters can act as the control plane for remote clusters (point to glossary entry for remote cluster).

I think the first one can replace the note about observing endpoints and facilitating cross-cluster LB, which seems a bit too much implementation detail of how they are connected, and not really part of the definition of a primary cluster.

File: content/en/docs/reference/glossary/remote-cluster.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+title: Remote Cluster
+test: n/a
+---
+
+A remote cluster is a cluster that uses a remote profile to connect to a
+control plane residing outside of the cluster. The control plane can reside
+either in another cluster or external to the mesh as is the case with managed
```
Comment (smawson) [inline content/en/docs/reference/glossary/remote-cluster.md:8]: We can link to the definition of primary cluster here right? A remote cluster can connect to a control plane running in a primary cluster or to a managed control plane.

We also have a glossary entry for managed control plane.

File: content/en/docs/reference/glossary/primary-cluster.md (hunk 1 above)
Comment (dcberg) [reply content/en/docs/reference/glossary/primary-cluster.md:7]: @frankbu is correct. A single istiod control plane can be configured to observe only a subset of a cluster.

Also the statement “which observes more than one cluster” is too strict. Is should be “which observes resources from one or more clusters...”.

File: content/en/docs/reference/glossary/primary-cluster.md (hunk 2 above)
Comment (dcberg) [reply content/en/docs/reference/glossary/primary-cluster.md:6]: I personally think the term “primary” is too vague. I prefer a name which is a bit more intention revealing such as “control cluster”.

File: content/en/docs/reference/glossary/primary-cluster.md (hunk 1 above)
Comment (nmittler) [reply content/en/docs/reference/glossary/primary-cluster.md:7]: @dcberg 

> @frankbu is correct. A single istiod control plane can be configured to observe only a subset of a cluster.

Just to be clear, are you referring to control plane revisions?

If we want to be more precise, maybe we can rephrase to something like:

```
A primary cluster is a cluster within a mesh that contains a control plane. The control plane observes configuration resources from a single source (e.g. Kubernetes API Server). It also observes service endpoints from one or more clusters in the mesh in order to facilitate cross-cluster load balancing between workloads.
```

File: content/en/docs/reference/glossary/primary-cluster.md (hunk 2 above)
Comment (frankbu) [reply content/en/docs/reference/glossary/primary-cluster.md:6]: I tend to agree "primary" is too vague, and so is "remote cluster" for that matter. Neither term really aligns very well with what they are. A primary cluster is a "self-controlled cluster" and a remote cluster is an "externally-controlled cluster" (the external control can come from the control plane in a self-controlled cluster or a control plane running outside of any cluster). I'm not sure that "self-controlled" and "externally-controlled" are very good names to use, but the point is that "primary" and "remote" don't really seem to be indicative of what they are.

File: content/en/docs/reference/glossary/primary-cluster.md (hunk 1 above)
Comment (frankbu) [reply content/en/docs/reference/glossary/primary-cluster.md:7]: > Multiple primary clusters can be connected together

I don't think what "connected together" means is clear. What's it trying to say? You can have more than one (synchronized) primary cluster in the same mesh for HA? I think it needs to say something more specific.

**Comment by frankbu** [comment]
@nmittler @smawson @dcberg @rcaballeromx 

Here's a proposal for both entries. WDYT? (maybe overkill on the links?)

```
---
title: Control Cluster
---

A control cluster is a [cluster](/docs/reference/glossary/#cluster) with a
[control plane](/docs/reference/glossary/#control-plane). You can have more than
one (synchronized) control cluster in a single [mesh](/docs/reference/glossary/#service-mesh)
for HA or to reduce latency. Control clusters can act as the control plane for
[remote clusters](/docs/reference/glossary/#remote-cluster).


---
title: Remote Cluster
---

A remote cluster is a [cluster](/docs/reference/glossary/#cluster that uses a
remote profile to connect to a [control plane](/docs/reference/glossary/#control-plane)
residing outside of the cluster. A remote cluster can connect to a control plane
running in a [control cluster](/docs/reference/glossary/#control-cluster)
or to a [managed control plane](/docs/reference/glossary/#managed-control-plane).
```

**Comment by smawson** [comment]
I don't like Control cluster and prefer Primary cluster.

**Comment by frankbu** [comment]
@smawson Two reasons I don't think "primary" is a great name is because of the definition of the word. It means either first or most important. So if you create a mesh using a managed control plane, the "first" cluster is not a primary cluster. Also, since you can have multiple primary clusters, they can't both be the "most important" one.

**Comment by smawson** [comment]
I had similar misgiving when it was initially proposed but I've mostly gotten past them because nothing better has been suggested.

Multi-primary is an established model in mysql at least: https://dev.mysql.com/doc/refman/8.0/en/group-replication-multi-primary-mode.html

The problem with Control cluster is that it has some negative connotations. Primary is also where K8s would like to head, and presumably they'll also run in multi-primary mode (for example with regional clusters).

For managed control planes there is still one cluster that has the config for the control plane, the control plane itself just doesn't run there. We can call that cluster the primary cluster (but not the control cluster!). Or we can call it the config cluster or something else.

**Comment by nmittler** [comment]
```
A control cluster is a [cluster](/docs/reference/glossary/#cluster) with a
[control plane](/docs/reference/glossary/#control-plane). You can have more than
one (synchronized) control cluster in a single [mesh](/docs/reference/glossary/#service-mesh)
for HA or to reduce latency. Control clusters can act as the control plane for
[remote clusters](/docs/reference/glossary/#remote-cluster).
```

Let's get rid of `(synchronized)`, I'm not sure that term has been defined and the concept isn't necessary here ... it's fine to have eventually or non-consistent primaries in the same mesh.

Otherwise, LG.

```
A remote cluster is a [cluster](/docs/reference/glossary/#cluster) that uses a
remote profile to connect to a [control plane](/docs/reference/glossary/#control-plane)
residing outside of the cluster. A remote cluster can connect to a control plane
running in a [control cluster](/docs/reference/glossary/#control-cluster)
or to a [managed control plane](/docs/reference/glossary/#managed-control-plane).
```

Let's get rid of `remote profile` here, it's an implementation detail. The first sentence can be simplified to `A remote cluster is a cluster that connects to a control plane residing outside of the cluster`. Also, have we defined `managed control plane` previously? I don't think we did for the concepts page.

**Comment by frankbu** [comment]
Updated proposal based on comments from @smawson and @nmittler 

@nmittler "managed control plane" is in the glossary, but I think you're right that it's not in the concepts.

```
---
title: Primary Cluster
---

A primary cluster is a [cluster](/docs/reference/glossary/#cluster) with a
[control plane](/docs/reference/glossary/#control-plane). You can have more than
one primary cluster in a single [mesh](/docs/reference/glossary/#service-mesh)
for HA or to reduce latency. Primary clusters can act as the control plane for
[remote clusters](/docs/reference/glossary/#remote-cluster).


---
title: Remote Cluster
---

A remote cluster is a [cluster](/docs/reference/glossary/#cluster that
connects to a [control plane](/docs/reference/glossary/#control-plane)
residing outside of the cluster. A remote cluster can connect to a control plane
running in a [primary cluster](/docs/reference/glossary/#primary-cluster)
or to a [managed control plane](/docs/reference/glossary/#managed-control-plane).
```

**Comment by nmittler** [comment]
@frankbu https://github.com/istio/istio.io/pull/7591#issuecomment-647780899 LGTM

**Comment by rcaballeromx** [comment]
@frankbu @nmittler @smawson I'll implement Frank's last suggestion then, agreed?

**Pull request merged**